src/main/java/com/icthh/xm/ms/entity/service/impl/FunctionServiceImpl.java

```diff
@@ -53,18 +60,15 @@ public FunctionContext execute(final String functionKey, final Map<String, Objec
         Objects.requireNonNull(functionKey, "functionKey can't be null");
         Map<String, Object> vInput = CustomCollectionUtils.emptyIfNull(functionInput);
 
-        FunctionSpec functionSpec = xmEntitySpecService.findFunction(functionKey).orElseThrow(
-            () -> new IllegalArgumentException("Function not found, function key: " + functionKey));
+        dynamicPermissionCheckService.checkContextPermission(DynamicPermissionCheckService.FeatureContext.FUNCTION,
```

Comment:
use static import for FUNCTION enum

Reply:
@sergeysenja1992 we already discussed it. Static imports potentially harmful. https://stackoverflow.com/questions/420791/what-is-a-good-use-case-for-static-import-of-methods
and advised to be used Very sparingly in official documentation. 
https://docs.oracle.com/javase/8/docs/technotes/guides/language/static-import.html

Reply:
Do not follow any articles thoughtlessly!
use your brain and common sense
In this case use static import!